Refactored ext:dev:upload and switched to Publisher API.

apascal07 opened the pull request.

joehan commented on src/commands/ext-dev-publish.ts:
```diff
@@ -16,14 +16,11 @@ marked.setOptions({
   .option(
     `--root <root>`,
-    `root directory that contains this Extension (defaults to previous version's root or root of repo if none set)`
+    `root directory that contains this extension (defaults to last uploaded root or repo root if none set)`
```
```suggestion
    `root directory that contains this extension (defaults to last uploaded root or "/" if none set)`
```

joehan commented on src/extensions/extensionsApi.ts:
```diff
@@ -19,7 +19,11 @@ import {
+const extensionsApiClient = new Client({ urlPrefix: extensionsOrigin, apiVersion: VERSION });
+const extensionsPublisherApiClient = new Client({
+  urlPrefix: extensionsOrigin,
+  apiVersion: VERSION,
```
Let's have separate consts for EXTENSIONS_API_VERSION and PUBLISHER_API_VERSION

joehan reviewed (commented):
Quick drive by review

apascal07 commented:
@joehan The diffing is awful in the tests file but I changed `publishExtensionVersion` to `createExtensionVersionFromLocalSource`, added `createExtensionVersionFromGitHubSource`, and deleted `unpublishExtension` (along with its method, there are no references to it anymore and it's been deprecated for ages).

joehan reviewed (approved):
LGTM
